AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginAuthCognitoTests/AWSDataStoreAuthBaseTest.swift

```diff
@@ -353,24 +309,24 @@ extension AWSDataStoreAuthBaseTest {
     ///   - onFailure: on failure callback
     func assertQuerySuccess<M: Model>(modelType: M.Type,
                                       _ expectations: AuthTestExpectations,
-                                      onFailure: @escaping (_ error: DataStoreError) -> Void) {
-        Amplify.DataStore.query(modelType).sink {
-            if case let .failure(error) = $0 {
-                onFailure(error)
+                                      onFailure: @escaping (_ error: DataStoreError) -> Void) async throws {
+        Task {
+            do {
+                let posts = try await Amplify.DataStore.query(modelType)
+                XCTAssertNotNil(posts)
+                await expectations.query.fulfill()
+            } catch {
+                onFailure(error as! DataStoreError)
             }
         }
-        receiveValue: { posts in
-            XCTAssertNotNil(posts)
-            expectations.query.fulfill()
-        }.store(in: &requests)
-        wait(for: [expectations.query],
-             timeout: 60)
+        await waitForExpectations([expectations.query], timeout: 60)
```

Comment:
this looks good! The task is created, so code waits on line 322, then when the awaited call in the task finishes (and does not hang) then `expectations.query` is fulfilled, causing line 322 to continue